File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/D_Remainder_Reminder.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll N, K;

void solve() {
  ll res = 0;
  for (ll b = K + 1; b <= N; ++b) {
    ll p = N / b;
    ll q = b - K;
    res += p * q + max(0LL, N - (p * b + K) + (K != 0));
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N >> K;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Hard100_2/python/D_Remainder_Reminder.py
N, K = map(int, input().split())
res = 0
for b in range(K + 1, N + 1):
    p = N // b
    q = b - K
    res += p * q + max(0, N - (p * b + K) + int(K != 0))
print(res)
